### ISSUE TITLE
could you provide a tutorial of how to train puzzle-mix

### ISSUE THREAD

**Issue by 774911840** [Westlake-AI/openmixup#22]
Hi,i want to train by puzzle-mix in this project,but i need some scripts.

**Comment by Lupin1998** [comment]
Hello, @774911840. You can train [PuzzleMix](https://arxiv.org/abs/2009.06962) by modifying `model.mix_mode="puzzlemix",` in config files for a mixup method. The only difference between PuzzleMix and other mixup methods is that PuzzleMix requires `soft Cross-Entropy` loss (set `use_soft=True` in `CrossEntropyLoss`). I provide two examples on CIFAR-100 and ImageNet. You can change the hyper-parameters in `model.mix_args.puzzlemix` according to its original paper (we provide the default setting).

(1) PuzzleMix on CIFAR-100 with ResNet-18. Modify [r18_puzzlemix_CE_soft.py](https://github.com/Westlake-AI/openmixup/blob/main/configs/classification/cifar100/mixups/basic/r18_mixups_CE_soft.py) as follows:
```python
_base_ = "r18_mixups_CE_none.py"

# model settings
model = dict(
    mix_mode="puzzlemix",  # modified
    head=dict(
        type='ClsMixupHead',  # mixup soft CE loss
        loss=dict(type='CrossEntropyLoss',  # soft CE (one-hot encoding)
            use_soft=True, use_sigmoid=False, loss_weight=1.0),
        with_avg_pool=True, multi_label=True, two_hot=False,
        in_channels=512, num_classes=100)
)
```

(2) PuzzleMix on ImageNet with ResNet-50. Modify [r50_mixups_CE_sigm_4xb64.py](https://github.com/Westlake-AI/openmixup/blob/main/configs/classification/imagenet/mixups/basic/r50_mixups_CE_sigm_4xb64.py) as follows:
```python
_base_ = "r50_mixups_CE_none_4xb64.py"

# model settings
model = dict(
    mix_mode="puzzlemix",  # modified
    head=dict(
        type='ClsMixupHead',  # mixup soft CE loss
        loss=dict(type='CrossEntropyLoss',  # soft CE (one-hot encoding)
            use_soft=True, use_sigmoid=False, loss_weight=1.0),
        with_avg_pool=True, multi_label=True, two_hot=False,
        in_channels=2048, num_classes=1000)
)

# additional hooks
custom_hooks = [
    dict(type='SAVEHook',
        save_interval=5004 * 10,  # plot every 10ep
        iter_per_epoch=5004),
]
```

**Comment by Lupin1998** [comment]
I close this issue if there is no more question. You can reopen it or start a new issue if you have more questions.
